src/hoverifier.ts

```diff
@@ -801,12 +823,134 @@ export function createHoverifier<C extends object, D, A>({
                 })
                 // Ensure the previously highlighted range is not highlighted and the new highlightedRange (if any)
                 // is highlighted.
-                const currentHighlighted = codeView.querySelector('.selection-highlight')
+                const currentHighlighted = codeView.querySelector(`.${selectionHighlightClass}`)
                 if (currentHighlighted) {
-                    currentHighlighted.classList.remove('selection-highlight')
+                    currentHighlighted.classList.remove(selectionHighlightClass)
                 }
                 if (hoveredTokenElement) {
-                    hoveredTokenElement.classList.add('selection-highlight')
+                    hoveredTokenElement.classList.add(selectionHighlightClass)
+                }
+            })
+    )
+
+    /**
+     * For every position, emits an Observable with new values for the `documentHighlightsOrError` state.
+     * This is a higher-order Observable (Observable that emits Observables).
+     */
+    const documentHighlightObservables: Observable<Observable<{
+        eventType: SupportedMouseEvent | 'jump'
+        dom: DOMFunctions
+        target: HTMLElement
+        adjustPosition?: PositionAdjuster<C>
+        codeView: HTMLElement
+        codeViewId: symbol
+        scrollBoundaries?: HTMLElement[]
+        documentHighlightsOrError?: typeof LOADING | DocumentHighlight[] | ErrorLike | null
+        position?: HoveredToken & C
+        part?: DiffPart
+    }>> = resolvedPositions.pipe(
+        map(({ position, codeViewId, ...rest }) => {
+            if (!position) {
+                return of({
+                    documentHighlightsOrError: null,
+                    position: undefined,
+                    part: undefined,
+                    codeViewId,
+                    ...rest,
+                })
+            }
+            // Get the document highlights for that position
+            return toMaybeLoadingProviderResult(getDocumentHighlights(position)).pipe(
+                catchError((error): [MaybeLoadingResult<ErrorLike>] => [{ isLoading: false, result: asError(error) }]),
```

Comment:
Are errors displayed in any way? If not, you may just wanna log here and return an empty array so you don't need the boilerplate to handle `ErrorLike`